### ISSUE TITLE
use Jobs when scheduling work

### ISSUE THREAD

**Issue by goern** [thoth-station/cleanup-job#12]
If we turn all workload some components schedule into Job rather than simple Pod, OpenShift will take care of cleaning them up.

The cleanup-job could be removed...

**Comment by fridex** [comment]
> The cleanup-job could be removed...

Is there a mechanism to stop a job after few tries? As we do not have a full control over parameters supplied, can we mark a job that cannot succeed as failed - e.g. the given image cannot be pulled?

**Comment by goern** [comment]
That's not it our job, the platform should take that work off of us ;)
Fridolín Pokorný <notifications@github.com> schrieb am Sa. 28. Apr. 2018 um
09:51:

**Comment by goern** [comment]
@fridex can you list the conditions that must be met to delete an old job? In general the job strives to succeed, if the job can pull and image, that might be an application level error, but not a job level error.

**Comment by fridex** [comment]
> @fridex can you list the conditions that must be met to delete an old job?

The current implementation waits 7 days. After that the given pod object is deleted from OpenShift.

> In general the job strives to succeed, if the job can pull and image, that might be an application level error, but not a job level error.

Ack, we can definitely do that. We will need to slightly redesign how we report status on API endpoint. Something like:

1.) check if the given analysis has an entry in the graph database
2.) check if the given pod has openshift status, if yes, what is the current status?

In this case we could report more detailed information about analysis status:
1. does not exist
2. it was scheduled, queued
3. it is running
4. it failed (e.g. image pull)
5. it succeeded, waiting for sync to the graph database
6. synced to the graph database

This way we can move to use purely jobs.

Now, we just report OpenShift pod status.

Just a detail - checks 1. and 2. can be done in parallel.

**Comment by fridex** [comment]
This is already done, let's close this issue.
